vault/expiration.go

```diff
@@ -332,12 +332,16 @@ func (m *ExpirationManager) RenewToken(req *logical.Request, source string, toke
 		return nil, err
 	}
 
-	// Fast-path if there is no renewal
 	if resp == nil {
 		return nil, nil
 	}
+
+	if resp.IsError() {
+		return resp, nil
```

Comment:
I think what we should do here is actually not trust what the backend gave us, but instead create a new logical.Response and copy the error from `resp` into the new one, then return that.